ethereum/p2p/src/main/java/org/hyperledger/besu/ethereum/p2p/peers/EnodeURL.java

```diff
@@ -40,22 +41,18 @@
 
   private final Bytes nodeId;
   private final InetAddress ip;
-  private final OptionalInt listeningPort;
-  private final OptionalInt discoveryPort;
+  private final Optional<Integer> listeningPort;
+  private final Optional<Integer> discoveryPort;
 
   private EnodeURL(
       final Bytes nodeId,
       final InetAddress address,
-      final OptionalInt listeningPort,
-      final OptionalInt discoveryPort) {
+      final Optional<Integer> listeningPort,
+      final Optional<Integer> discoveryPort) {
     checkArgument(
         nodeId.size() == NODE_ID_SIZE, "Invalid node id.  Expected id of length: 64 bytes.");
-    checkArgument(
-        !listeningPort.isPresent() || NetworkUtility.isValidPort(listeningPort.getAsInt()),
-        "Invalid listening port.  Port should be between 1 - 65535.");
-    checkArgument(
-        !discoveryPort.isPresent() || NetworkUtility.isValidPort(discoveryPort.getAsInt()),
-        "Invalid discovery port.  Port should be between 1 - 65535.");
+    listeningPort.ifPresent(port -> NetworkUtility.checkPort(port, "listening"));
```

Comment:
Will exceptions be thrown out of this lambda construct?  We should have a unit test to validate, all the unit tests I reviewed are all within UInt16.

Reply:
Is it not higher here?

https://github.com/RatanRSur/besu/blob/5720fdcbcfb887e4255de37add482f0c3ef4fa0b/ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/peers/EnodeURLTest.java#L276

Reply:
I was scanning for the builder variants (`.discoveryPort(int)` and `.listeningPort(int)`.)  Are there negative port tests?  We should test those too.  Possibly via the builders.